collections/_posts/2023-03-08-Variationnal_Dropout_and_the_local_reparameterization_trick.md

```diff
@@ -0,0 +1,243 @@
+- the multiplicative noise determined by $$\alpha$$ 
+
+
+> During dropout training, $$\theta$$ is adapted to maximize the expected log likelihood  $$\mathbb{E}_{q_{\alpha}} \left[ L_{\mathcal{D}}(\theta) \right] $$. 
+For this to be consistent with the optimization of a variational lower bound, 
+the prior on the weights $$p(\textbf{w})$$ has to be such that $$D_{KL}(q_\phi(\textbf{w}) \vert \vert p(\textbf{w}))$$ does not depend on $$\theta$$.
+
+They show that the only prior possible is :
+
+$$p(log(\vert w_{i,j} \vert)) \propto c$$
+
+Therefore, they try to maximize :
+
+$$\mathbb{E}_{q_{\alpha}} \left[ L_{\mathcal{D}}(\theta) \right] - D_{KL}(q_\alpha(\textbf{w}) \vert \vert p(\textbf{w}))$$
+
+- $$\theta$$ and $$\alpha$$ are treated as hyperparameters that are fixed during training.
+
+However, $$- D_{KL}(q_\alpha(\textbf{w}) \vert \vert p(\textbf{w}))$$ is anatycally intractable but can be approximated by :
+
+$$- D_{KL} \left[ q_\phi(w_i) \vert \vert p(w_i) \right] \approx constant + 0.5log(\alpha) + c_1\alpha + c_2\alpha^2 + c_3\alpha^3$$
+
+with
+
+$$ c_1 = 1.16145124, c_2 = -1.50204118 , c_3 = 0.58629921$$
+
+Now that there is a derived dropout's variational objective, 
+they show that maximizing the variational lower bound with respect to $$\alpha$$ will make the hyperparameters $$\alpha$$ and $$\theta$$ adaptative.
+Furthermore, it will be possible to learn a separate dropout rate per layer, per neuron, of even per separate weight.
+
+Despite this new approach, they have found that large values of $$\alpha$$ will imply large-variance gradients. 
+That's why they apply a constraint on $$\alpha$$ ($$\alpha \le 1$$ i.e.* $$p \le 0.5$$) during training.
+
+We will show in future post on sparse variational dropout that this constraint can be removed and induce sparse representation of a neural network.
+
+&nbsp;
+
+
+# Results
+
+Comparison of their method to standard binary dropout and two versions of Gaussian dropout.
+- Gaussian dropout A : pre-linear Gaussian dropout (based on a paper)
+- Gaussian dropout B : post-linear Gaussian dropout (based on another paper)
+
+Two types of variational dropout as well :
+- Variational dropout A : correlated weight noise (equivalent of an adaptative Gaussian dropout type A)
+- Variational dropout A2 denotes performance of type A variational dropout but with a KL-divergence downscaled with a factor of 3 (seems to prevent underfitting)
+- Variational dropout B : independant weight noise (equivalent of Gaussian dropout B)
+
+Tests were done on MNIST dataset for a classification task. 
+They choosed the same architecture from a publication as well as the same drop rate which correspond to $$p \le 0.5$$ for the hidden layers and $$p \le 0.2$$ for the input layer.
+
+![](/collections/images/variational_dropout/img1.jpg)
+
+
+&nbsp;
+
+
+# Conclusion
+
+- They have obtain an efficient estimator with a low computational complexity and a low variance by injecting noise locally instead of globally.
```

Comment:
Conjugation:
```suggestion
- They obtained an efficient estimator with a low computational complexity and a low variance by injecting noise locally instead of globally.
```